src/main/java/dev/sigstore/fulcio/client/SigningCertificate.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2022 The Sigstore Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.sigstore.fulcio.client;
+
+import com.google.api.client.util.PemReader;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.Gson;
+import com.google.gson.JsonParseException;
+import dev.sigstore.json.GsonSupplier;
+import java.io.*;
+import java.security.cert.*;
+import java.util.ArrayList;
+import java.util.Base64;
+import org.conscrypt.ct.DigitallySigned;
+import org.conscrypt.ct.SerializationException;
+import org.conscrypt.ct.SignedCertificateTimestamp;
+
+/** Response from Fulcio that includes a certPath and an SCT */
+public class SigningCertificate {
+
+  private final CertPath certPath;
+  private final SignedCertificateTimestamp sct;
+
+  static SigningCertificate newSigningCertificate(String certs, String sctHeader)
+      throws CertificateException, IOException, SerializationException {
+    CertPath certPath = decodeCerts(certs);
+    SignedCertificateTimestamp sct = decodeSCT(sctHeader);
+    return new SigningCertificate(certPath, sct);
+  }
+
+  @VisibleForTesting
+  static CertPath decodeCerts(String content) throws CertificateException, IOException {
+    PemReader pemReader = new PemReader(new StringReader(content));
+    CertificateFactory cf = CertificateFactory.getInstance("X.509");
+    ArrayList<X509Certificate> certList = new ArrayList<>();
+    while (true) {
+      PemReader.Section section = pemReader.readNextSection();
+      if (section == null) {
+        break;
+      }
+
+      byte[] certBytes = section.getBase64DecodedBytes();
+      certList.add((X509Certificate) cf.generateCertificate(new ByteArrayInputStream(certBytes)));
+    }
+    if (certList.isEmpty()) {
+      throw new IOException("no certificates were found in response from Fulcio");
+    }
+
+    return cf.generateCertPath(certList);
+  }
+
+  @VisibleForTesting
+  static SignedCertificateTimestamp decodeSCT(String sctHeader) throws SerializationException {
+    byte[] sct = Base64.getDecoder().decode(sctHeader);
+    Gson gson = new GsonSupplier().get();
+    return gson.fromJson(new InputStreamReader(new ByteArrayInputStream(sct)), SctJson.class)
```

Comment:
I am not sure what SCT is, however, it looks like something should declare the encoding that is used to store SCT

```suggestion
    return gson.fromJson(new InputStreamReader(new ByteArrayInputStream(sct), StandardCharsets.UTF_8), SctJson.class)
```

Reply:
SCT are "signed certificate timestamps", they are authenticated metadata used to verify that a certifciate was logged with a certificate transparency log. The incoming SCT format is kind of fulcio specific, but it's definitely utf-8.

Reply:
I do think this will be changed to an embedded sct format in https://github.com/sigstore/fulcio/pull/507, but this is currently how in prod fulcio works